ui/static/ui/js/manage_taxonomies.jsx

```diff
@@ -6,24 +6,116 @@ define('manage_taxonomies', ['react', 'lodash', 'jquery', 'utils'],
   var ICheckbox = Utils.ICheckbox;
 
   var TermComponent = React.createClass({
+    mixins: [React.addons.LinkedStateMixin],
+    getInitialState: function() {
+      return {
+        formatActionClassName: 'fa fa-pencil',
```

Comment:
It looks like this depends entirely on `formatActionState`, you should remove it from state and calculate it in `render` instead based on the value of `formatActionState`